clippy_utils/src/consts.rs

```diff
@@ -400,6 +400,85 @@ impl<'a, 'tcx> ConstEvalLateContext<'a, 'tcx> {
         let res = self.typeck_results.qpath_res(qpath, id);
         match res {
             Res::Def(DefKind::Const | DefKind::AssocConst, def_id) => {
+                // Check if this constant is based on `cfg!(..)`,
+                // which is NOT constant for our purposes.
+                if let Some(node) = self.lcx.tcx.hir().get_if_local(def_id) {
+                    fn node_contains_cfg_expn(tcx: TyCtxt<'_>, n: Node<'_>) -> bool {
```

Comment:
Not sure if this should live somewhere else. It could maybe be useful to others as as more generic `clippy_utils::contains_expn_of(e: &Expr, name: &str) -> bool`. Or it could just live in the `assertions_on_constants` lint.